wildguard/wildguard.py

```diff
@@ -179,7 +179,12 @@ def __init__(
         if ephemeral_model:
             self.model = None
         else:
-            self.model = LLM(model=MODEL_NAME)
+            gpu_name = torch.cuda.get_device_name(0)
+            if gpu_name == 'NVIDIA GeForce RTX 3090':
```

Comment:
```suggestion
            if torch.cuda.get_device_properties(0).total_memory < 30e9:
```

And to make this work with `ephemeral_model=True`, you will also need to add a parameter for `max_model_len` to `subprocess_inference_with_vllm` and pass it through to `create_and_inference_with_vllm`.